### ISSUE TITLE
Dialyzer warning when using generator with MIX_ENV=dev

### ISSUE THREAD

**Issue by evnu** [alfert/propcheck#45]
I am trying to make generators available in the `dev` environment. We use `dialyxir` to check for type discrepancies. The following minimal example creates a dialyzer warning:

```elixir
# lib/dial.ex
defmodule Dial do
  use PropCheck

  def test do
    oneof([1, 2])
  end
end
```

The warning:

```
lib/dial.ex:4: Function test/0 has no local return
lib/dial.ex:5: The call 'Elixir.PropCheck.BasicTypes':oneof([1 | 2,...]) does not have a term of type ['Elixir.PropCheck.BasicTypes':raw_type(),...] (with opaque subterms) as 1st argument
```

Only two dependencies are needed to create this warning:

```elixir
# mix.exs
  defp deps do
    [
      {:propcheck, "~>1.0"},
      {:dialyxir, "~> 0.5", only: [:dev], runtime: false},
    ]
  end
```

**Comment by evnu** [comment]
To work around this issue, I need to manually disable the warning:

```
defmodule Dial do
  use PropCheck

  @dialyzer {:no_opaque, test: 0}
  @dialyzer {:no_return, test: 0}

  def test do
    oneof([1])
  end
end
```

**Comment by evnu** [comment]
I tried to reproduce this in plain Erlang, but no success so far. The following does not generate a dialyzer warning:

```erlang
-module(proper_dialyzer).

-export([gen/0]).

-include_lib("proper/include/proper.hrl").

gen() ->
    oneof([1, 2]).
```

I generated the type spec dialyzer derives with `typer`:

```sh
typer --plt _build/default/rebar3_21.0.3_plt -I _build/default/lib/ -I ./_build/default/lib/proper/include -pa _build/default/lib/proper/ebin _build/default/lib/proper/src/* src/proper_dialyzer.erl
```

Resulting spec:

```
%% ... snipping most of the output for  PropEr modules ...

-spec oneof([raw_type(),...]) -> proper_types:type().

%% File: "src/proper_dialyzer.erl"
%% -------------------------------
-spec gen() -> proper_types:type().
```

So, for my example above, dialyzer derived the expected type for the generator.

The example above can be found in https://github.com/evnu/proper-dialyzer as a rebar3 project.

**Comment by alfert** [comment]
Could you please give the current HEAD a try? I changed a lot type specs in #61 which may positively impact your issue.

**Comment by evnu** [comment]
* Dialyxir 1.0.0-rc3 on top of PropCheck 1.0.0-rc1:

```
lib/dial.ex:4:no_return
Function test/0 has no local return.
________________________________________________________________________________
lib/dial.ex:5:call_without_opaque
The call PropCheck.BasicTypes.oneof([1 | 2,...]) does not have a term of type ['Elixir.PropCheck.BasicTypes':raw_type(),...] (with opaque subterms) in 1.
________________________________________________________________________________
done (warnings were emitted)
```

* Dialyxir 1.0.0-rc3 on top of PropCheck dd61d3dd11d899a3f9107eec49c3a0801fb18ab2: **no warnings**

This looks great, thank you! We will be moving to Dialyxir 1.0.0 then. Can you release PropCheck on hex.pm with the changes from HEAD?

**Comment by evnu** [comment]
> I will do so

Great, thanks! It seems that 1.1.0 was pushed to master, but is not yet present on hex.pm?

**Comment by alfert** [comment]
Enjoy the 1.1.0 release on hex.pm!
